### DISCUSSION TITLE
pgrid sensor modbus

### DISCUSSION THREAD

**Discussion by okanano** [mletenay/home-assistant-goodwe-inverter#275] [Q&A] [answered]
Hi, I am connected through modbus serial RTU to my inverter GoodWe model GW5048D-ES. I'm mapping the registers but not sure if the register address for pgrid (Hex 0518 - Dec  1304)is correct because I'm getting different values with Home Assistant add-on specifically when is charging, or is this sensor calculated? If yes, could you please provide me the formula?

**Comment by mletenay** [comment, marked as answer]
'pgrid' in HA is calculated, resp. its sign is.
See https://github.com/marcelblijleven/goodwe/blob/0feabc41031c7f320aad8ef837032956e9d5cce9/goodwe/es.py#L68

It takes (absolute) power value from reg 0x518 and sign from reg 0x052F (01 - export to grid, 02 - import from grid)

**Reply by okanano** [reply]
thanks @mletenay !!
